src/validation.cpp

```diff
@@ -5737,6 +5737,16 @@ bool ChainstateManager::DeleteSnapshotChainstate()
     return true;
 }
 
+ChainstateRole Chainstate::GetRole() const
+{
+    if (m_chainman.GetAll().size() <= 1) {
```

Comment:
288dd732b5c0085618598fe3dbba23ac6d194173

I wish the chainstate state machine was briefly mentioned here, or referenced where it's defined.
A reviewer needs a way to understand why it's impossible to have 2+ chainstates, one of them being NORMAL (and others being something).

Reply:
Added a pointer to the ChainstateManager documentation this.